### ISSUE TITLE
Bug: The `name` value is not propagated in the extended configs

### ISSUE THREAD

**Issue by TrevinAvery** [typescript-eslint/typescript-eslint#10086]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

```TypeScript
No code needed. Just run eslint with the below config and it will error as expected.
```

### ESLint Config

```javascript
import eslint from '@eslint/js'
import tseslint from 'typescript-eslint'

export default tseslint.config(
	{
		name: "TypeScript",
		files: ['**/*.{ts,tsx}'],
		extends: [
                        // intentional bug here, these items should use the spread oporator because they are arrays
			tseslint.configs.strictTypeChecked,
			tseslint.configs.stylisticTypeChecked,
		],
		languageOptions: {
			parserOptions: {
				projectService: true,
				tsconfigRootDir: __dirname, // use import.meta.dirname after updating to Node 20+
			},
		},
	},
)
```

### Expected Result

When an error occurs with one of the configs, I expect the output to include the `name` of the config, but when using `extends`, the configs are all "unnamed".

### Actual Result

In the example above I have added the name `TypeScript`. But when it errors I see 
> ConfigError: Config (**unnamed**): Unexpected key "0" found.

### Additional Info

The issue appears to be [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/src/config-helper.ts#L100-L103). It only adds `files` and `ignores` to the extended objects. This should also included the name appended with an index. For example

```js
...(config.name && { name: `${config.name}-${index}` }),
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/rule-tester`       | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |
